src/public/index.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  CSVUploader,
+  CSVEntityPicker,
+  CSVEntityName,
+  CSVEntityProperties,
+  CSVEdgeMapping,
+  CSVImportFeedback,
+} from "./cards";
+import { EntitiesTypes } from "./models";
+import * as utils from "./utils";
+
+function main() {
+  let entityType: EntitiesTypes;
+
+  /************** Initialize plugin  ************/
+
+  const uploader = new CSVUploader();
+  uploader.init();
+
+  const entityPicker = new CSVEntityPicker();
+  entityPicker.init();
+
+  const entityName = new CSVEntityName();
+  entityName.init();
+
+  const entityProperties = new CSVEntityProperties();
+  entityProperties.init();
+
+  const edgeMapping = new CSVEdgeMapping();
+  edgeMapping.init();
+
+  const importFeedback = new CSVImportFeedback();
+  importFeedback.init();
+
+  /************** Set event handlers ************/
+
+  // cancel button (go to first page and reset state)
+  const cancelButtons = document.getElementsByClassName(
+    "cancelButton"
+  ) as HTMLCollectionOf<HTMLElement>;
+  for (let i = 0; i < cancelButtons.length; i++) {
+    cancelButtons[i].addEventListener("click", () => {
+      resetPlugin();
+    });
+  }
+
+  // first screen event handler
```

Comment:
why not adding the event handler in the classes ?

Reply:
this makes all classes independent from each other and avoid the need to add some kind of dependency injection system or a bus messaging to share states. We can discuss more if you want, but for a small project it makes things easier

Reply:
here index.ts has access to the only instance of each class and controls all the flow